blivet/partitioning.py

```diff
@@ -564,17 +564,10 @@ def do_partitioning(storage):
             storage.devicetree._remove_device(partition, modparent=False, force=True)
 
     partitions = storage.partitions[:]
-    for part in storage.partitions:
-        part.req_bootable = False
-        if not part.exists:
-            # start over with flexible-size requests
-            part.req_size = part.req_base_size
 
-    try:
-        storage.boot_device.req_bootable = True
-    except AttributeError:
-        # there's no stage2 device. hopefully it's temporary.
-        pass
+    if hasattr(storage, "set_boot"):
+        # set the boot flag on boot device
+        storage.set_boot()
```

Comment:
I'm sorry to say this, but I don't like these ``hasattr()`` calls. I have no great idea how to replace them, but maybe by defining ``set_bool()`` doing nothing in the ``Blivet`` class? Or maybe add ``cares_about_bootloader (prop)`` to the ``Blivet`` class, override it in the ``InstallerStorage`` and check that?